app/src/pages/inside/logsPage/logsGrid/flexibleDateTime/flexibleDateTime.jsx

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import { dateFormat, getMicroSeconds } from 'common/utils/timeDateUtils';
+import { LOG_TIME_FORMAT_ABSOLUTE, LOG_TIME_FORMAT_EXTENDED } from 'controllers/user/constants';
+import { useDispatch, useSelector } from 'react-redux';
+import { setLogTimeFormatAction, userIdSelector } from 'controllers/user';
+import { logTimeFormatSelector } from 'controllers/user/selectors';
+import {
+  getLogTimeFormatFromStorage,
+  setLogTimeFormatInStorage,
+} from 'controllers/log/storageUtils';
+import styles from './flexibleDateTime.scss';
+
+const cx = classNames.bind(styles);
+
+export const FlexibleDateTime = ({ time, customClass }) => {
+  const dispatch = useDispatch();
+  const logTimeFormat = useSelector(logTimeFormatSelector);
+  const userId = useSelector(userIdSelector);
+
+  useEffect(() => {
+    const format = getLogTimeFormatFromStorage(userId);
+    dispatch(setLogTimeFormatAction(format));
+  }, []);
+
+  const isAbsolute = () => logTimeFormat === LOG_TIME_FORMAT_ABSOLUTE;
+  const toggleFormat = () => {
+    const format = isAbsolute() ? LOG_TIME_FORMAT_EXTENDED : LOG_TIME_FORMAT_ABSOLUTE;
+    dispatch(setLogTimeFormatAction(format));
+    setLogTimeFormatInStorage(userId, format);
+  };
+  const absoluteTime = dateFormat(time);
+
+  const extractMilliseconds = () => getMicroSeconds(time).slice(0, 3);
+
+  const extractMicroSeconds = () => getMicroSeconds(time).slice(3);
+
+  const extractTime = () => absoluteTime.slice(-8);
+  return (
+    <button
+      className={cx('abs-rel-time', { extended: !isAbsolute() }, customClass)}
```

Comment:
Let's rename the component to f.e. `FlexibleLogTime` as we use log specific naming there. Also adjust the class name accordinely.